### PR TITLE
Identify template parts in error messages

### LINKED ISSUE #28056: Improve the "Template Part Not Found" error message
**Is your feature request related to a problem? Please describe.**
When a full site editing template part is not found, the displayed message is "Template Part Not Found".
This is not so helpful when there is more than one template part not found.

**Describe the solution you'd like**
It would be great if the message could also include information about which template it expected.

### PR DESCRIPTION
**Pull request by carolinan**
## Description

Fixes https://github.com/WordPress/gutenberg/issues/28056

This PR adds the template part slug to the error messages when a template part can not be found.
The error message on the front is only visible to logged in users and is now translatable.

## How has this been tested?

To confirm the issue in https://github.com/WordPress/gutenberg/issues/28056 you first need a full site editing theme where a template part is missing.
You can install TT1 Blocks and rename either the header or footer template part files inside
the block-template-parts folder, and view the site editor and the front.

In the site editor, you should see the message: Template part has been deleted or is unavailable.
On the front, you should see: Template Part Not Found.
If you log out, you should see that the error message is visible to all visitors.

Apply the PR. 
View the front while logged out. The error message should not be visible.
Login. The error message should now say: Template part has been deleted or is unavailable: (template part slug)
Open the site editor. The error message should now say: Template part has been deleted or is unavailable: (template part slug)

## Screenshots 
template part error message with slug to identify which template part that is missing.

## Types of changes

Enhancement, bug fix.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/template-part/edit/index.js [hunk 1]
```diff
@@ -70,8 +70,12 @@ export default function TemplatePartEdit( {
 		return (
 			<TagName { ...blockProps }>
 				<Warning>
-					{ __(
-						'Template part has been deleted or is unavailable.'
+					{ sprintf(
+						/* translators: %s: template part slug */
```
Comment (aristath) [inline packages/block-library/src/template-part/edit/index.js:74]: Nitpicking: Should "Template" be capitalized here?
```suggestion
						/* translators: %s: Template part slug */
```

File: packages/block-library/src/template-part/index.php [hunk 2]
```diff
@@ -51,8 +51,12 @@ function render_block_core_template_part( $attributes ) {
 		}
 	}
 
-	if ( is_null( $content ) ) {
-		return 'Template Part Not Found';
+	if ( is_null( $content ) && is_user_logged_in() ) {
+		return sprintf(
+			/* translators: %s: template part slug. */
```
Comment (aristath) [inline packages/block-library/src/template-part/index.php:56]: Same as above, maybe "Template" should be capitalized?
```suggestion
			/* translators: %s: Template part slug. */
```

**Review by aristath — CHANGES_REQUESTED** [review]

**Review by aristath — APPROVED** [review]
LGTM 👍

File: packages/block-library/src/template-part/index.php [hunk 3]
```diff
@@ -65,8 +65,12 @@ function render_block_core_template_part( $attributes ) {
 		}
 	}
 
-	if ( ! $template_part_id || is_null( $content ) ) {
-		return __( 'Template Part not found.' );
+	if ( is_null( $content ) && is_user_logged_in() ) {
```
Comment (carolinan) [inline packages/block-library/src/template-part/index.php:68]: Should this check for a user role and not only logged in users?

Comment (aristath) [reply packages/block-library/src/template-part/index.php:68]: I was thinking about that too... Maybe `current_user_can( edit_themes );` 🤔

Comment (carolinan) [reply packages/block-library/src/template-part/index.php:68]: Perhaps leave it as is until https://github.com/WordPress/gutenberg/issues/27597

Comment (aristath) [reply packages/block-library/src/template-part/index.php:68]: ^ That's why I approved the PR 😆

**Pull request merged**